From 93698e5c6ce450480a1911bbbdb159718fad73d3 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 29 Nov 2024 18:23:30 +0100
Subject: [PATCH 1515/1695] plat-stm32mp1: remove unused function
 pm_cb_is_valid

Remove the unused function pm_cb_is_valid().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e0fc904a04df63f34ae792f1a171e60a6573b9c1 plat-stm32mp1: remove unused function pm_cb_is_valid]
Fixes: cf483e4784e5 ("plat-stm32mp1: add low power drivers for stm32mp1")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/423277
Change-Id: I76395eefc76ddcb6ae963f7b9330034d78014b56
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/437073
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/context.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/pm/context.c b/core/arch/arm/plat-stm32mp1/pm/context.c
index 85324b27c..b4ea04852 100644
--- a/core/arch/arm/plat-stm32mp1/pm/context.c
+++ b/core/arch/arm/plat-stm32mp1/pm/context.c
@@ -221,25 +221,6 @@ static void restore_time(void)
 #endif
 }
 
-static bool __maybe_unused pm_cb_is_valid(void (*cb)(enum pm_op op, void *hdl),
-					  void *hdl)
-{
-	void *cb_voidp = (void *)(vaddr_t)cb;
-	paddr_t cb_phy = virt_to_phys(cb_voidp);
-	paddr_t hdl_phy = virt_to_phys(hdl);
-	bool valid = false;
-
-	valid = (phys_to_virt(cb_phy, MEM_AREA_TEE_RAM_RX, 1) == cb_voidp) &&
-		((phys_to_virt(hdl_phy, MEM_AREA_TEE_RAM_RX, 1) == hdl) ||
-		 (phys_to_virt(hdl_phy, MEM_AREA_TEE_RAM_RO, 1) == hdl) ||
-		 (phys_to_virt(hdl_phy, MEM_AREA_TEE_RAM_RW, 1) == hdl));
-
-	if (!valid)
-		EMSG("pm_cb mandates unpaged arguments %p %p", cb_voidp, hdl);
-
-	return valid;
-}
-
 uintptr_t stm32mp_pm_retram_resume_ep(void)
 {
 	struct retram_resume_ctx *ctx = get_retram_resume_ctx();
-- 
2.47.2

